assets/src/story-embed-block/block/edit.js

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useState, useEffect, useRef } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { ResizableBox } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import EmbedControls from './embedControls';
+import EmbedLoadinng from './embedLoading';
+import EmbedPlaceholder from './embedPlaceholder';
+import StoryPlayer from './storyPlayer';
+import { icon } from './index.js';
+import './edit.css';
+
+const MIN_SIZE = 20;
+
+function StoryEmbedEdit({ attributes, setAttributes, className, isSelected }) {
+  const {
+    url: outerURL,
+    width = 360,
+    height = 600,
+    align = 'none',
+    poster,
+    title,
+  } = attributes;
+
+  const [editingURL, setEditingURL] = useState(false);
+  const [localURL, setLocalURL] = useState(outerURL);
+  const [isFetchingData, setIsFetchingData] = useState(false);
+  const [storyData, setStoryData] = useState({});
+  const [cannotEmbed, setCannotEmbed] = useState(false);
+
+  const showLoadingIndicator = isFetchingData;
+  const showPlaceholder = !localURL || !outerURL || editingURL || cannotEmbed;
+
+  const ref = useRef();
+
+  useEffect(() => {
+    setLocalURL(outerURL);
+  }, [outerURL]);
+
+  useEffect(() => {
+    if (ref.current && global.ampStoryPlayer) {
+      const player = new global.ampStoryPlayer(global, ref.current);
+      player.load();
+    }
+  }, [showLoadingIndicator, showPlaceholder]);
+
+  const fetchStoryData = useCallback(
+    (url) => {
+      if (!url) {
+        return;
+      }
+
+      try {
+        setIsFetchingData(true);
+        const urlObj = new URL(url);
+
+        apiFetch({
+          path: `web-stories/v1/embed?url=${urlObj.toString()}`,
+        })
+          .then((data) => {
+            setCannotEmbed(!data?.title);
+            setStoryData(data);
+            setAttributes({
+              url: localURL,
+            });
+          })
+          .catch((data) => {
+            setStoryData(data);
```

Comment:
Why are you doing here?

Reply:
In the case of an error, `data` contains an error object from the REST API that we can use to show an error to the user. `storyData?.message` is passed to `EmbedPlaceholder` because of this.